crates/core/src/db/relational_db.rs

```diff
@@ -154,7 +168,7 @@ impl RelationalDB {
     /// of the database in case of an incomplete replay.
     ///
     /// This restriction may be lifted in the future to allow for "live" followers.
-    pub fn apply<T>(self, history: T) -> Result<Self, DBError>
+    pub fn apply<T>(self, history: T) -> Result<(Self, ConnectedClients), DBError>
```

Comment:
Similarly, I'd add to the docs:

```rs
/// Watches for `__identity_connected__` / `__identity_disconnected__` calls in the log
/// in order to construct a [`ConnectedClients`],
/// the set of clients who were connected as of the most recent TX,
/// so that the newly-constructed module can call disconnect for each of them.
```